File: Javascript/Car.js
class Car {
  constructor(license, driver) {
    this.id;
    this.license = license;
    this.driver = driver;
    this.passenger;
  }
  printDataCar = () => {
    console.table(this.driver);
    console.log(this.driver.name);
    console.log(this.driver.document);
  };
  printDataCarU = (uber) => {
    console.table(uber);
  };
}
/* con este podemos crear un metodo a
partir de la function Car
*/
// Car.prototype.printDataCar = function () {
//   console.table(this.driver);
//   // mostrara una tabla con todos los datos

//   console.log(this.driver.name);
//   console.log(this.driver.document);
//   // console.log("License: " + this.license + " Name Driver: " + this.driver);
// };

File: Java/Car.java
class Car {
    Integer id;
    String license;
    // se modifica el tipo de dato a Account para heredar los datos
    Account driver;
    // modificador de acceso si no tiene nada toma el modificador como default
    private Integer passenger;

    public Car(String license, Account driver) {
        this.license = license;
        this.driver = driver;
    }

    void printDataCar() {
        if (passenger != null) {
            System.out
                    .println("License: " + license + " Name Driver: " + driver.name + " passengers: " + passenger);
        }
    }

    public Integer getPassenger() {
        return passenger;
    }

    void setPassenger(Integer passenger) {
        if (passenger == 4) {
            this.passenger = passenger;
        } else {
            System.out.println("Necesitas asignar 4 pasajeros");
        }
    }

}